Sources/Clappr/Classes/Base/Loader.swift

```diff
@@ -1,63 +1,46 @@
 open class Loader {
-    internal(set) open var playbackPlugins: [Plugin.Type] = [AVFoundationPlayback.self, NoOpPlayback.self]
-    #if os (iOS)
-    internal(set) open var containerPlugins: [Plugin.Type] = [PosterPlugin.self, SpinnerPlugin.self]
-    #else
-    internal(set) open var containerPlugins: [Plugin.Type] = []
-    #endif
-    internal(set) open var corePlugins = [Plugin.Type]()
 
-    internal(set) open var mediaControl: MediaControl.Type = MediaControl.self
+    public static let shared = Loader()
+    public var plugins: [String: Plugin.Type] = [:]
 
-    fileprivate var externalPlugins = [Plugin.Type]()
-
-    public convenience init() {
-        self.init(externalPlugins: [])
+    var playbacks: [Plugin.Type] {
+        return plugins.filter { $0.value.type == .playback }.map { return $0.value }
```

Comment:
There's a slight issue here regarding the playbacks. A `Playback` isn't a regular plugin since only a single one is loaded for a given `Container`. Another behavior that's different is that it's loaded on demand depending on the source we need to play and that there should be a fixed order in which the playbacks are checked.

For these reasons, I belive it's better to have the playbacks actually separated from the actual plugins. So instead of using a computed property here, we should have the playbacks stored in its own stored property, as an array.

Given that change, the `Loader` will need to have a separate `registerPlayback` method, and some other changes. Since it's a big change, we could have it done in a later PR.

Reply:
I totally agree, we will create another PR with these changes.
We have other PRs for the Media Control that need the code developed here, are you guys ok to approve and analyse the next PR with the last comments made here?